src/webots/nodes/utils/WbNodeUtilities.cpp

```diff
@@ -819,7 +819,7 @@ WbRobot *WbNodeUtilities::findRobotAncestor(const WbNode *node) {
   if (!node)
     return NULL;
 
-  while (node->parentNode()) {
+  while (node) {
```

Comment:
After this change, the function name is no longer consistent with the functionality.

I didn't check why this change is needed but previously the check if the node was a robot was done outside this function. So we should be consistent and keep the function as it was before, or change the name of the function and adapt the previous calls of this function.

Reply:
Without the fix, if the robot node is the root (it always used to be Group) this function fails to find the robot ancestor.

But you are right, maybe I should detect that edge case instead of doing a quick fix.

Reply:
Actually, it doesn't change the behavior, but only fixes the edge case. The behavior and the function don't match in any case (if I am not missing something). If the passed node is a robot it will return the node itself.

Reply:
It changes the behavior because if the node is robot but doesn't have any robot ancestor previously it was returning NULL and now it returns the node.
A function that searches for ancestors (as described by the name) is not meant to detect if the node is a robot or not. Note that this check is almost always done externally before calling this function.

Reply:
Oh, I see, I thought you were pointing at something else

Reply:
Actually, I'm wrong. This is only the condition that assumes that every nodes as a parent (top parent is Group). So the behavior is unchanged.